modules/Bio/EnsEMBL/Variation/VCFCollection.pm

```diff
@@ -486,59 +487,87 @@ sub get_all_VariationFeatures_by_Slice {
+                  my $tv = Bio::EnsEMBL::Variation::TranscriptVariation->new_fast({
+                    variation_feature  => $vf,
+                    _feature_stable_id => $transcript_map{'Feature'}
+                  });  
+                  #Add variation Feature seq from transcript map
+                  my $tva = Bio::EnsEMBL::Variation::TranscriptVariationAllele->new_fast({
+                    variation_feature_seq => $transcript_split[0],
+                    is_reference               => 0
+                  });
+                  my @cons_list = split('&',$transcript_map{'Consequence'});
+                  foreach my $con (@cons_list){
+                    if ( exists $OVERLAP_CONSEQUENCES{$con} ) {
+                      my $new_cons = $OVERLAP_CONSEQUENCES{$con};
+                      $tva->add_OverlapConsequence($new_cons);
+                    }
+                    else{
+                      print("The consequence is not available:",$con,"\n");
+                    }
                   }
+                  $tv->add_TranscriptVariationAllele($tva);
+                  $vf->add_TranscriptVariation($tv);
                 }
-                $tv->add_TranscriptVariationAllele($tva);
-                $vf->add_TranscriptVariation($tv);
               }
             }
           }
         }
+        else{ 
+          $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
+            -variation_feature  => $vf,
+            -no_ref_check       => 1,
+          );
+          weaken($vf->{intergenic_variation}->{base_variation_feature});
+        }
+        $vf->_finish_annotation();
       }
-      else{ 
-        $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
-          -variation_feature  => $vf,
-          -no_ref_check       => 1,
-        );
-        weaken($vf->{intergenic_variation}->{base_variation_feature});
+    } elsif ($vcf_info_field eq 'CLNSIG'){
+      foreach my $vf (@vfs) {
+        my $info = $vf->{vcf_record}->get_info;
+        my $vcf_variants_mix_case = $info->{$vcf_info_field};
+        my $vcf_variants = lc($vcf_variants_mix_case);
+
+        # Replace commas for specific examples that contain a comma,
+        # otherwise they would split/break these examples - comma is also a delimiter
+        # Not ideal as new comma-containing terms will need to be added as they appear in 
+        # list found in Bio::EnsEMBL::Variation::Utils::Config::clinvar_clinical_significance_types
+        $vcf_variants =~ s/pathogenic,_low_penetrance/pathogenic_low_penetrance/;
+        $vcf_variants =~ s/likely_pathogenic,_low_penetrance/likely_pathogenic_low_penetrance/;
+
+        my @clnsig_vars = split('[\|,/;]',$vcf_variants);
+        foreach my $clnsig_var (@clnsig_vars) {
+          $clnsig_var =~ s/_/ /g;
+          $clnsig_var =~ s/^\s+|\s+$//;
+          if (grep /^$clnsig_var$/, @Bio::EnsEMBL::Variation::Utils::Config::clinvar_clinical_significance_types){
```

Comment:
Instead of calling directly `clinvar_clinical_significance_types` you can get it from `%ATTRIBS`:
```
my $clinsig_list = $ATTRIBS{clinvar_clin_sig};
if (grep /^$clnsig_var$/, @{$clinsig_list}) {
```

Reply:
Thanks for this, it is much cleaner now!